test/tools/ossfuzz/protoToAbiV2.cpp

```diff
@@ -0,0 +1,730 @@
+
+	// Add typed params for calling public and external functions with said type
+	appendTypedParams(
+		CalleeType::PUBLIC,
+		isValueType(_dataType),
+		_type,
+		_paramName,
+		((m_varCounter == 1) ? Delimiter::SKIP : Delimiter::ADD)
+	);
+	appendTypedParams(
+		CalleeType::EXTERNAL,
+		isValueType(_dataType),
+		_type,
+		_paramName,
+		((m_varCounter == 1) ? Delimiter::SKIP : Delimiter::ADD)
+	);
+}
+
+void ProtoConverter::visitArrayType(std::string const& _baseType, ArrayType const& _x)
+{
+	std::string type = arrayTypeAsString(_baseType, _x);
+	std::string varName, paramName;
+	createDeclAndParamList(type, DataType::ARRAY, varName, paramName);
+	// Resize-initialize array and add checks
+	resizeInitArray(_x, _baseType, varName, paramName);
+}
+
+void ProtoConverter::visitType(
+	DataType _dataType,
+	std::string const& _type,
+	std::string const& _value
+)
+{
+	std::string varName, paramName;
+	createDeclAndParamList(_type, _dataType, varName, paramName);
+	addCheckedVarDef(_dataType, varName, paramName, _value);
+}
+
+void ProtoConverter::appendVarDeclToOutput(
+	std::string const& _type,
+	std::string const& _varName,
+	std::string const& _qualifier
+)
+{
+	// One level of indentation for state variable declarations
+	// Two levels of indentation for local variable declarations
+	if (m_isStateVar)
+		m_output << Whiskers(R"(
+	<type><?qual> <qualifier></qual> <varName>;)"
+		)
+		("type", _type)
+		("qual", !_qualifier.empty())
+		("qualifier", _qualifier)
+		("varName", _varName)
+		.render();
+	else
+		m_output << Whiskers(R"(
+		<type><?qual> <qualifier></qual> <varName>;)"
+		)
+		("type", _type)
+		("qual", !_qualifier.empty())
+		("qualifier", _qualifier)
+		("varName", _varName)
+		.render();
+}
+
+void ProtoConverter::appendChecks(
+	DataType _type,
+	std::string const& _varName,
+	std::string const& _rhs
+)
+{
+	std::string check = {};
+	switch (_type)
+	{
+	case DataType::STRING:
+		check = Whiskers(R"(!bytesCompare(bytes(<varName>), <value>))")
+			("varName", _varName)
+			("value", _rhs)
+			.render();
+		break;
+	case DataType::BYTES:
+		check = Whiskers(R"(!bytesCompare(<varName>, <value>))")
+			("varName", _varName)
+			("value", _rhs)
+			.render();
+		break;
+	case DataType::VALUE:
+		check = Whiskers(R"(<varName> != <value>)")
+			("varName", _varName)
+			("value", _rhs)
+			.render();
+		break;
+	case DataType::ARRAY:
+		solUnimplemented("Proto ABIv2 fuzzer: Invalid data type.");
+	}
+
+	// Each (failing) check returns a unique value to simplify debugging.
+	m_checks << Whiskers(R"(
+		if (<check>) return <returnVal>;)"
+	)
+		("check", check)
+		("returnVal", std::to_string(m_returnValue++))
+		.render();
+}
+
+void ProtoConverter::addVarDef(std::string const& _varName, std::string const& _rhs)
+{
+	// State variables cannot be assigned in contract-scope
+	// Therefore, we buffer state variable assignments and
+	// render them in function scope later.
+	if (m_isStateVar)
+		m_statebuffer << Whiskers(R"(
```

Comment:
You could also first render the whiskers template to a local variable of string type and then add it either to `m_statebuffer` or `m_output` depending on `m_isStateVar`.